spec/index.bs

```diff
@@ -7575,31 +7579,47 @@ GPUQueue includes GPUObjectBase;
 
             **Returns:** {{undefined}}
 
-            - If |source|.{{GPUImageCopyExternalImage/source}}
-                <l spec=html>[=is not origin-clean=]</l>,
+            1. Let |sourceImage| be |source|.{{GPUImageCopyExternalImage/source}}
+            1. Run [=Check the usability of the image argument=](|sourceImage|).
+                If it throws an exception, stop.
+                If it does not return `good`, throw an {{InvalidStateException}} and stop.
+            1. If |sourceImage| <l spec=html>[=is not origin-clean=]</l>,
                 throw a {{SecurityError}} and stop.
-
-            If any of the following requirements are unmet, throw an {{OperationError}} and stop.
-            <div class=validusage>
-                - Let |textureDesc| be |destination|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/[[descriptor]]}}.
-                - |copySize|.[=Extent3D/depthOrArrayLayers=] must be `1`.
-                - |textureDesc|.{{GPUTextureDescriptor/usage}} must include both {{GPUTextureUsage/RENDER_ATTACHMENT}} and {{GPUTextureUsage/COPY_DST}}.
-                - |textureDesc|.{{GPUTextureDescriptor/dimension}} must be {{GPUTextureDimension/"2d"}}.
-                - |textureDesc|.{{GPUTextureDescriptor/format}} must be one of the following:
-                    - {{GPUTextureFormat/"rgba8unorm"}}
-                    - {{GPUTextureFormat/"rgba8unorm-srgb"}}
-                    - {{GPUTextureFormat/"bgra8unorm"}}
-                    - {{GPUTextureFormat/"bgra8unorm-srgb"}}
-                    - {{GPUTextureFormat/"rgb10a2unorm"}}
-                    - {{GPUTextureFormat/"rgba16float"}}
-                    - {{GPUTextureFormat/"rgba32float"}}
-                    - {{GPUTextureFormat/"rg8unorm"}}
-                    - {{GPUTextureFormat/"rg16float"}}
-            </div>
-
-            Issue: The above list was written just for ImageBitmap.
-            Re-evaluate it for canvas (and video).
-            (It's probably still fine, though `rgb10a2unorm` is a bit of an outlier.)
+            1. If any of the following requirements are unmet, throw an {{OperationError}} and stop.
+                <div class=validusage>
+                    - |source|.|origin|.[=Origin3D/x=] + |copySize|.[=Extent3D/width=]
+                        must be &le; the width of |sourceImage|.
+                    - |source|.|origin|.[=Origin3D/y=] + |copySize|.[=Extent3D/height=]
+                        must be &le; the height of |sourceImage|.
+                    - |source|.|origin|.[=Origin3D/z=] + |copySize|.[=Extent3D/depthOrArrayLayers=]
+                        must be &le; 1.
```

Comment:
What's the rational for allowing a depth of `0`? That's a departure from the validation logic this is replacing. Also, this implicitly prevents copying to texture array layers other than the 0th layer, which I'm not sure is intended?

Seems like the desired logic may be something along the lines of simply:
```
- |copySize|.[=Extent3D/depthOrArrayLayers=] must be 1;
```

Reply:
This only prevents copying *from* layers other than the 0th (the source image is 2d so always has only one layer).

This behavior is more consistent with copyTextureToTexture: it validates the same as a copy from a 2d texture with one layer.